sky/backends/backend_utils.py

```diff
@@ -916,6 +916,11 @@ def write_cluster_config(
             tmp_yaml_path,
             cluster_config_overrides=to_provision.cluster_config_overrides)
         kubernetes_utils.combine_metadata_fields(tmp_yaml_path)
+        valid, message = kubernetes_utils.check_pod_config(
+            tmp_yaml_path, dryrun)
```

Comment:
Running this on a fresh kubernetes cluster which does not already contain `skypilot-service-account` fails:
```
W 12-18 22:00:56 cloud_vm_ray_backend.py:2065] sky.exceptions.ResourcesUnavailableError: Failed to provision on cloud Kubernetes due to invalid cloud config: sky.exceptions.InvalidCloudConfigs: There are invalid config in pod_config, deatil: pods "Unknown" is forbidden: error looking up service account default/skypilot-service-account: serviceaccount "skypilot-service-account" not found
```

Note that this service account is created in our downstream provisioning logic in config.py before the pod is provisioned. We may want to move this check there.